src/screens/Authentication/AdminSignin/AdminSignin.tsx

```diff
@@ -25,20 +23,40 @@ function AdminSigninScreen({
   const [email, setEmail] = useState('');
   const [password, setPassword] = useState('');
   const { signInEmail } = useContext(AuthContext);
+  const [emailError, setEmailError] = useState('');
+  const [ signInError, setSignInError ] = useState('');
 
   const onSubmit: SubmitHandler<FormValues> = async data => {
     try {
+      signInWithEmailAndPassword(auth, email, password)
+      .catch(error => {
+        setSignInError("Oops! Incorrect email or password. Try again.");
+      });
+      emailSchema.parse(email);
       await signInEmail(email, password);
-      console.log('signed in');
       // navigation.navigate('Root', { screen: 'Home' });
     } catch (e) {
-      console.error(e);
+      if (e instanceof z.ZodError) {
+        setEmailError("Oops! Invalid email. Try again.");
+        console.log(e.issues);
+      }
+      console.log(e.message);
       throw e;
     }
   };
 
-  const onError: SubmitErrorHandler<FormValues> = (errors, e) =>
-    console.log(errors);
+  const handleEmailChange = (email) => {
+    setEmail(email);
+    setEmailError('');
```

Comment:
same comment as before. only need to reset error messages if they're not already empty strings